internal/pkg/validator/validator.go

```diff
@@ -59,3 +75,31 @@ func processValidateError(err validator.ValidationErrors) error {
 
 	return result.ErrorOrNil()
 }
+
+func validateTemplateInputId(fl validator.FieldLevel) bool {
+	return regexpcache.MustCompile(`^[a-zA-Z\.\_]+$`).MatchString(fl.Field().String())
+}
+
+// Default value must be of the same type as the Type.
+func validateTemplateInputDefault(fl validator.FieldLevel) bool {
+	if fl.Field().Kind() == reflect.Ptr && fl.Field().IsNil() {
+		return true
+	}
+	if fl.Field().IsZero() {
+		return true
+	}
+	return fl.Field().Kind().String() == fl.Parent().FieldByName("Type").String()
+}
```

Comment:
@webrouse Tak s tím budu asi potřebovat poradit 😕 Dokonce i když sem dám jen `return true`, tak test pořád failuje:
![CleanShot 2021-12-14 at 11 08 47](https://user-images.githubusercontent.com/215660/145978466-354c375a-dd6c-4830-90c4-f66962ab9aa8.jpg)

Je to tenhle test: https://github.com/keboola/keboola-as-code/blob/3c9cedbd4047c0841d9f44edc6d9819e203be42f/internal/pkg/template/input_test.go#L108-L117
`default` v něm chybí, což by mělo být ok protože nemá nastavený `required`: https://github.com/keboola/keboola-as-code/blob/3c9cedbd4047c0841d9f44edc6d9819e203be42f/internal/pkg/template/input.go#L14

Reply:
Ked pridasna zaciatok `omitempty`? Tj. nevalidovat prazdnu hodnotu?
Len tipujem, ked nepomoze, zavolame.

Reply:
Ha, to přeskočí validaci úplně? Tak to by mělo pomoct, zkusím.

Reply:
Mylim, ze je to tak nejak vymyslene a nieco podobne som uz riesil ... ale nie som si uplne isty.

Reply:
Aha, já měl `omitempty` v `json` (tam funguje tak, že pokud není vyplněný, tak se negeneruje do jsonu, že?) ale nedošlo mi že se dá přidat i do `validate` a to fakt pomohlo. 👍

Reply:
jj, je to tak.